PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,11 @@
+###### Explanation About What Code Achieves:
+<!-- Please explain why this code is necessary / what it does -->
```

Comment:
I tend to like [the format used by Twitter's Finagle project](https://github.com/twitter/finagle/blob/develop/.github/PULL_REQUEST_TEMPLATE.md), which first frames the problem that the change sets about to solve and then describes the chosen solution. This has the benefit of encouraging contributors to think concretely about what problem they're solving, rather than making changes for change's sake.

In this model, the solution section could include a description of the testing done.

Reply:
Update to use that model~ Thanks!